### PR TITLE
Add confirmation dialog for Remove All and New File

### PR DESCRIPTION
**Pull request by pdesaulniers**
Implements https://github.com/falkTX/Carla/issues/360. It prompts for confirmation even if there is only one plugin loaded. 

I hope I did that right :)

### CODE REVIEW

File: source/carla_host.py [hunk 1]
```diff
@@ -1008,6 +1008,11 @@ def slot_pluginRemoveAll(self):
         if self.fPluginCount == 0:
             return
 
+        ask = QMessageBox.question(self, self.tr("Remove All"), self.tr("Are you sure you want to remove all plugins?"), QMessageBox.Yes|QMessageBox.No)
```
Comment (falkTX) [inline source/carla_host.py]: multiple lines here please, and no need to save the return value, just compare directly to the No option.

**Comment by falkTX** [comment]
open to thought, should the dialog be opened even if only 1 plugin is loaded?
perhaps only on >= 2?

also, is there any code in carla calling `slot_removeAll()` that would trigger this warning when not wanted?
(like changing projects in NSM)

**Comment by pdesaulniers** [comment]
>  is there any code in carla calling slot_removeAll() that would trigger this warning when not wanted?

Ah, yes, it's called by the "New File" and "Open" actions. It's sort of appropriate for the New File action, but not for Open.

File: source/carla_host.py (hunk 1 above)
Comment (pdesaulniers) [reply source/carla_host.py]: Maybe it's just me, but I think this looks a bit less clear without saving the return value:
```
if QMessageBox.question(self, self.tr("Remove All"), 
                              self.tr("Are you sure you want to remove all plugins?"), 
                              QMessageBox.Yes|QMessageBox.No) == QMessageBox.No:
                                            return
```

**Comment by pdesaulniers** [comment]
> should the dialog be opened even if only 1 plugin is loaded? perhaps only on >= 2?

This wouldn't be a huge time-saver :p 

By showing the dialog all the time, you get consistent behavior, and can possibly save a bit of frustration.

File: source/carla_host.py (hunk 1 above)
Comment (falkTX) [reply source/carla_host.py]: I still prefer this version tbh

**Comment by pdesaulniers** [comment]
Seems OK now. ```slot_fileNew ``` seems to be only called by the menu action. Same thing for the new ```slot_confirmRemoveAll```.

File: source/carla_host.py [hunk 2]
```diff
@@ -1003,6 +1011,18 @@ def slot_pluginAdd(self):
...
+
+        if QMessageBox.question(self, self.tr("Remove All"),
+                                      self.tr("Are you sure you want to remove all plugins?"),
+                                      QMessageBox.Yes|QMessageBox.No) == QMessageBox.No:
+                                        return
+
+        self.slot_pluginRemoveAll()
+
     @pyqtSlot()
     def slot_pluginRemoveAll(self):
```
Comment (falkTX) [inline source/carla_host.py]: Since this function is no longer being used as a connection slot, better to remove the decoration and prefix from it.

**Comment by falkTX** [comment]
Looks great!
Perhaps I would change the wording on those questions, but a really minor thing and can be done later anytime.
Thanks!

**Pull request merged**